classes/class.ilObjAdobeConnectGUI.php

```diff
@@ -748,9 +748,17 @@ public function performSso()
 
                     $presentation_url = ilAdobeConnectServer::getPresentationUrl();
 
-                    $xmlAPI->logout( $_SESSION['xavc_last_sso_sessid'] );
                     //login current user session
-                    $session = $ilAdobeConnectUser->loginUser();
+					if( !ilAdobeConnectServer::getSetting('enhanced_security_mode') == false || $settings->getAuthMode() == ilAdobeConnectServer::AUTH_MODE_DFN)
```

Comment:
I am not a fan of adding this `if/elseif/else` conditions. This should be requested polymorphically on `$xmlAPI`.

Reply:
done